generics/action-steps.js

```diff
@@ -1,140 +1,48 @@
+    When(/^user enters "(.*)" into the "(.*)" field$/, async (text, textField) => {
         const fieldID = textField.replace(/ /g, '_').toUpperCase();
-
         pageMap[global.pageID][fieldID].clear();
-        pageMap[global.pageID][fieldID].sendKeys(text).then(next);
+        await pageMap[global.pageID][fieldID].sendKeys(text);
     });
 
-    When(/^user selects "(.*)" from the "(.*)" drop down$/, (text, buttonText, next) => {
+    When(/^user selects "(.*)" from the "(.*)" drop down$/, async (text, buttonText) => {
         const buttonID = buttonText.replace(/ /g, '_').toUpperCase();
-        pageMap[global.pageID][buttonID].sendKeys(text).then(next);
-        // element(by.xpath(`//option[normalize-space()="${text}"]`)).click().then(next);
+        await pageMap[global.pageID][buttonID].sendKeys(text);
     });
 
-
-    When(/^user enters values into the following fields$/, (table, next) => {
+    When(/^user enters values into the following fields$/, async (table) => {
         const hash = table.rowsHash();
         const keys = Object.keys(hash);
-        let valuesEntered = 0;
         for (let i = 0; i < keys.length; i++) {
             const fieldID = keys[i].replace(/ /g, '_').toUpperCase();
             try {
                 pageMap[global.pageID][fieldID].clear();
-                pageMap[global.pageID][fieldID].sendKeys(hash[keys[i]]).then((result) => {
-                    if (result) {
-                        // console.log(`\x1b[31m\t${keys[i]} -> ${hash[keys[i]]} \x1b[0m`);
-                        return next(result);
-                    }
-                    driver.actions().sendKeys(webdriver.Key.TAB).perform().then((result) => {
-                        valuesEntered++;
-                        // console.log(`\x1b[32m\t${keys[i]} -> ${hash[keys[i]]} \x1b[0m`);
-                        if (valuesEntered === keys.length) next();
-                    });
-                });
+                const result = await pageMap[global.pageID][fieldID].sendKeys(hash[keys[i]]);
+                if (result) return result;
+                await driver.actions().sendKeys(webdriver.Key.TAB).perform();
             }
-            catch(err) {
+            catch (err) {
                 throw new Error(`Error sending keys to the ${fieldID} element of ${global.pageID}`);
             }
-
         }
     });
 
-    When(/^user presses the (.*) key/, (key, next) => {
-        expect(driver.actions().sendKeys(webdriver.Key[key]).perform()).to.eventually.be.ok.and.notify(next);
+    When(/^user presses the (.*) key/, async (key) => {
+        await driver.actions().sendKeys(webdriver.Key[key]).perform();
     });
 
-    Then(/^user waits for ([0-9]+) seconds$/, async (num) => {
+    When(/^user waits for ([0-9]+) seconds$/, async (num) => {
         await driver.sleep(num * 1000);
     });
-    //
-    // Then(/^user waits for "([^"]*)" element to be present$/, { timeout: 2 * 60 * 1000 }, (elementText, next) => {
```

Comment:
go rid of the commented code that seems to be not useful anymore.

let me know if i should keep any of it, my personal asses is to not keep it, most likely we will not need it in the future as we have not had any major issues in the executions that used abacus.